src/Filter/Strategy/FilterStrategy.php

```diff
@@ -2,15 +2,26 @@
 
 namespace Webfactory\VisibilityFilterBundle\Filter\Strategy;
 
+use Webfactory\VisibilityFilterBundle\Filter\ParameterCollection;
+
 /**
  * Implement this to apply custom filtering logic to your project.
  */
 interface FilterStrategy
 {
+    /**
+     * Adds all parameters to the collection that affect the resulting filter SQL string. If the filter SQL depends on
+     * factors other than parameters added to the collection, the cache won't be invalidated, which will have undesired
+     * consequences!
+     *
+     * So please add all parameters to the collection.
```

Comment:
"undesired consequences" and "please" seem too vague for my taste. Would you mind explaining it in more detail (or simply remove the unprecise wording)?

Reply:
> which will have undesired consequences!

suggestion: which will have undesired consequences such as objects being visible for users who should not see them. (no acclamation mark)

> Adds all parameters to the collection

seems to contradict 

> So please add all parameters to the collection.

Either the method adds all parameters or the user has to do it. Maybe "Adds _the_ parameters" would be better, as it avoids using the identical wording.

> So please add all parameters to the collection.

Suggestion: To avoid this, make sure that all parameters are added to the collection.